### ISSUE TITLE
Error creating native barcode detector (Error with 4.2.2 only)

### ISSUE THREAD

**Issue by hsaidi** [nisrulz/qreader#23]
I am having this error : NativeBarcodeDetectorHandle: Error creating native barcode detector
Because of that i got this one: QREader: Barcode recognition libs are not downloaded and are not operational

This error is happening with Android 4.2.2 only, the same code runs perfectly fine with 5.1 and 6.0.

Device: smartwatch with a full install of android.
The sample code to detect the QrCodes hasn't been modified.

No instant run.

**Comment by nisrulz** [comment]
@hsaidi the library was never tested for android wear. Anyways from what you have mentioned in the issue, its straightforward that the library is not able to download the required mobile vision library code , which is provided by google. If you are running android in your smartwatch and its not connected to internet / it doesnot have the capability to run google play services , it wont work.

**Comment by hsaidi** [comment]
Well, the smartwatch has a full version of android, not android wear.
And when it worked, on the 5.1 and 6.0 version, the device wasn't connected to internet.

**Comment by nisrulz** [comment]
Can you confirm your 4.2.2 version has google play services installed.

Explain your scenario as to how you are able to test it on multiple version of Android on the smartwatch. Are you using the same smart watch to format and flash each version of Android on it  ? 

This is un-expected way of running android on smartwatch, in anycase the expected would have been android wear, I can only guess here, as I can't replicate this. 

I could however have tested it on an Android Wear emulator image, if it was the case.

**Comment by hsaidi** [comment]
Yes, installed and running.
I have a two phones and a smartwatch on which i am testing apps.

The same code works on 6.0 ad 5.1. on the smartwatch, which is really a smartphone with a small screen as it has everything a smartphone has and is running a full version of android as i said. the code doesn't work, the camera opens, the problem lies at the .init() of the barcode detector.

The smartwatch comes with android 4.2. it was made to run the full version of android as opposed to other smartwatchs that run android wear.

**Comment by nisrulz** [comment]
Ok, can you clear your cache and confirm you have enough space on the device so that it can download the extra library files it depends on ? 

Checkout this [stackoverflow answer](https://stackoverflow.com/questions/32099530/google-vision-barcode-library-not-found) too.

**Comment by nisrulz** [comment]
@hsaidi any updates on this issue ? Did you solve it ?

**Comment by hsaidi** [comment]
I had to use a smartwatch with android 5 on it. couldn't make it work with the 4.2.2

**Comment by nisrulz** [comment]
Updated the library to v2.0, with a lot of fixes and API cleanup. Please give it a try and let me know if this issue was fixed.

Thank you

**Comment by alexmelyon** [comment]
@nisrulz please write in readme.md that Google Services are required.
It will no work in Genymotion, at least.

**Comment by nisrulz** [comment]
@alexmelyon Good point. Even though the library says that it uses Google's mobile vision API, it would be good to have it specified in the readme as to what is required for it to work. I will do the required. I hope to sit down soon and fix a lot of issues this lib is having.

**Comment by nisrulz** [comment]
Already added to readme.
